include/libopencm3/pac55xx/memorymap.h

```diff
@@ -0,0 +1,77 @@
+/** @file
+ * @defgroup memorymap
+ *
+ * Peripheral Memory Map for the PAC55xx series of microcontrolelrs.
+ *
+ * @ingroup PAC55xx
+ * @author Brian Viele <vielster@allocor.tech>
+ * LGPL License Terms @ref lgpl_license
+ * @date 1 Dec 2019
+ */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef INCLUDE_LIBOPENCM3_PAC55XX_MEMORYMAP_H_
+#define INCLUDE_LIBOPENCM3_PAC55XX_MEMORYMAP_H_
+
+/* Memory and Peripheral memory Mapping */
+
+/** @defgroup address_memory Address Memory Map.
+@ingroup memorymap
+@{*/
+#define PAC55XX_FLASH_BASE          (0x00000000UL)
+#define PAC55XX_INFO1_FLASH_BASE    (0x00100000UL)
+#define PAC55XX_INFO2_FLASH_BASE    (0x00100400UL)
+#define PAC55XX_INFO3_FLASH_BASE    (0x00100800UL)
+#define PAC55XX_SRAM_BASE           (0x20000000UL)
+#define PAC55XX_PERIPH_BASE         (0x40000000UL)
+/**@}*/
+/** @defgroup peripheral_addresses Core Peripheral Memory Map.
+@ingroup memorymap
+@{*/
+#define PAC55XX_ADC_BASE            (PAC55XX_PERIPH_BASE + 0x00000)
+#define PAC55XX_I2C_BASE            (PAC55XX_PERIPH_BASE + 0x10000)
+#define PAC55XX_USARTA_BASE         (PAC55XX_PERIPH_BASE + 0x20000)
+#define PAC55XX_USARTB_BASE         (PAC55XX_PERIPH_BASE + 0x30000)
+#define PAC55XX_USARTC_BASE         (PAC55XX_PERIPH_BASE + 0x40000)
+#define PAC55XX_USARTD_BASE         (PAC55XX_PERIPH_BASE + 0x50000)
+#define PAC55XX_TIMERA_BASE         (PAC55XX_PERIPH_BASE + 0x60000)
+#define PAC55XX_TIMERB_BASE         (PAC55XX_PERIPH_BASE + 0x70000)
+#define PAC55XX_TIMERC_BASE         (PAC55XX_PERIPH_BASE + 0x80000)
+#define PAC55XX_TIMERD_BASE         (PAC55XX_PERIPH_BASE + 0x90000)
+#define PAC55XX_CAN_BASE            (PAC55XX_PERIPH_BASE + 0xA0000)
+#define PAC55XX_GPTIMERA_BASE       (PAC55XX_PERIPH_BASE + 0xB0000)
+#define PAC55XX_GPTIMERB_BASE       (PAC55XX_PERIPH_BASE + 0xC0000)
+#define PAC55XX_SYS_PERIPH_BASE     (PAC55XX_PERIPH_BASE + 0xD0000)
```

Comment:
I couldn't put my finger on it earlier, but I see now :)  None of these need a PAC55XX_ prefix.  these _are_ the peripherals, all the peripheral files will start with the same prefix as well, and  you're never including two memorymaps at the same time, as the dispatch header prevents it.  It doesn't _hurt_ per se here, but you can skip them.

Reply:
Done.